feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md

```diff
@@ -0,0 +1,145 @@
+# PF-1.2: Policy-based traffic GRE Encapsulation to IPv4 GRE tunnel
+
+## Summary
+
+The test verifies policy forwarding(PF) encapsulation action to IPv4 GRE tunnel when matching on source/destination.
+
+## Testbed type
+
+*  [`featureprofiles/topologies/atedut_4.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_4.testbed)
+
+## Procedure
+
+### Test environment setup
+
+*   DUT has an ingress port and 2 egress ports.
+
+    ```
+                             |         | --eBGP-- | ATE Port 2 |
+        [ ATE Port 1 ] ----  |   DUT   |          |            |
+                             |         | --eBGP-- | ATE Port 3 |
+    ```
+
+*   Traffic is generated from ATE Port 1.
+*   ATE Port 2 is used as the destination port for encapsulated 
+    traffic.
+*   ATE Port 3 is used as the fallback destination for
+    pass-through traffic.
+
+#### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces. Configure MTU of 9216 (L2) on ATE Port1, MTU 2000 on ATE Port 2, 3
+ 
+2.  IPv4 and IPv6 static routes to test destination networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  Another set of IPv4 static routes to 32x IPv4 GRE encap destinations towards ATE Port 2.
+
+4.  2 IPv4 and 2 IPv6 source prefixes will be used to generate tests traffic 
+(SRC1-SRC2). Apply policy-forwarding with 4 rules to DUT Port 1:
+    - Match IPV4-SRC1 and accept/foward.
+    - Match IPV6-SRC1 and accept/foward.
+    - Match IPV4-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+    - Match IPV6-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+
+5.   Set GRE encap source to device's loopback interface.
+6.   Either `identifying-prefix` or `targets/target/config/destination` can be used to configure GRE destinations.
```

Comment:
How should the test automation code decide which one to use?  Is this a deviation based on the NOS of the DUT?  (junos vs others?). I'm kind of surprised to see this in the OC model, but to get through this test given the current model, the README should specify a method to chose which path to use.